src/charm.py

```diff
@@ -318,6 +320,11 @@ def _on_collect_status(self, event: CollectStatusEvent) -> None:  # noqa: C901
                 BlockedStatus(f"Missing integration {HYDRA_ENDPOINTS_INTEGRATION_NAME}")
             )
 
+        if not oauth_integration_exists(self):
+            event.add_status(
+                BlockedStatus(f"Missing integration {HYDRA_ENDPOINTS_INTEGRATION_NAME}")
```

Comment:
This suggests that the `hydra-endpoint-info` integration does not exist, could it point to missing `oauth` relation instead?
```suggestion
                BlockedStatus(f"Missing integration {OAUTH_INTEGRATION_NAME}")
```

Reply:
Good catch, fixed it. Fixed the test that was supposed to catch this as well.